openff/bespokefit/executor/services/optimizer/worker.py

```diff
@@ -20,12 +21,34 @@
 @celery_app.task(bind=True, acks_late=True)
 def optimize(self, optimization_input_json: str) -> str:
 
+    from openff.toolkit.typing.engines.smirnoff import ForceField
+
     input_schema = parse_raw_as(
         Union[BespokeOptimizationSchema], optimization_input_json
     )
     input_schema.id = self.request.id
 
-    optimizer = get_optimizer(input_schema.optimizer.type)
-    result = optimizer.optimize(input_schema, keep_files=False)
+    input_force_field = ForceField(input_schema.initial_force_field)
+
+    stage_results = []
+
+    for stage in input_schema.stages:
```

Comment:
I might have missed if this is handled in the coordinator somewhere but I think we should just be expecting a `OptimizationStageSchema` and perform one optimisation? or is the idea to do all tasks for that stage at once? So when the schema gets here all stages have already had their reference data generated? If so I think the qcgenerator /fragmentation stages will need to be updated to do all tasks for every optimisation stage as well.